flaml/searcher/flow2.py

```diff
@@ -555,25 +565,46 @@ def suggest(self, trial_id: str) -> Optional[Dict]:
                         if self.step > self.step_ub:
                             self.step = self.step_ub
             else:
-                # check if the new config is different from self.best_config
+                # check if the new config is different from best_config
                 same = True
                 for key, value in config.items():
-                    if key not in self.best_config or value != self.best_config[key]:
+                    if key not in best_config or value != best_config[key]:
                         same = False
                         break
                 self._same = same
         if self._num_proposedby_incumbent == self.dir and (
                 not self._resource or self._resource == self.max_resource):
             # check stuck condition if using max resource
             self._num_proposedby_incumbent -= 2
-            self._init_phrase = False
+            self._init_phase = False
             if self.step >= self.step_lower_bound:
                 # decrease step size
                 self._oldK = self._K if self._K else self._iter_best_config
                 self._K = self.trial_count_proposed + 1
                 self.step *= np.sqrt(self._oldK / self._K)
             else:
                 return None
+        if self._init_phase:
+            return unflatten_dict(config)
+        if self._trunc == 1 and self._direction_tried is not None:
+            # random
+            for i, key in enumerate(self._tunable_keys):
+                if self._direction_tried[i] != 0:
+                    for _, generated in generate_variants({'config': {
+                        key: self.space[key]
+                    }}):
+                        if generated['config'][key] != best_config[key]:
+                            config[key] = generated['config'][key]
+                            return unflatten_dict(config)
+                        break
+        else:
+            # check if config == best_config
```

Comment:
What is this for?

Reply:
If the rounded config is equal to the best config, then the incumbent would not be moved. By moving the incumbent, we increase the chance to escape that local point.